src/Xamarin.Android.Build.Tasks/Utilities/ManifestDocument.cs

```diff
@@ -578,6 +578,11 @@ XElement CreateApplicationElement (XElement manifest, string applicationClass, L
 			List<UsesConfigurationAttribute> usesConfigurationAttr = [];
 			foreach (var assemblyPath in Assemblies) {
 				var assembly = Resolver.GetAssembly (assemblyPath);
+				if (assembly == null) {
+					log.LogDebugMessage ($"Assembly '{assemblyPath}' not found.");
+					continue;
+				}
+
```

Comment:
I think I'd like to keep it, it's a good safeguard for situations when something goes really wrong - at least we have a clear message in the log, instead of a NREX